### DISCUSSION TITLE
[DNA damage]: Scoring single strand break (SSB) and double-strand breaks (DSB).

### DISCUSSION THREAD

**Discussion by thanhtaiphys** [topas-nbio/TOPAS-nBio#21] [Simulation-question]
### Simulation question

Hello everyone,

I am starting to learn how to use TOPAS to compare the DNA damage effects of Ac-225 and Lu-177, specifically focusing on **single-strand breaks (SSB)** and **double-strand breaks (DSB)**.

To do this, I have learned that we could use a cell model (including cell and nucleus) and then use DBSCAN to score DSB.

In addition, from reading the TOPAS-nBio documentation, I understand that we can use the **LinearDNA model** or the **Nucleus model** and then use _DNADamageStepByStep_ to score.

Given your expertise, I would greatly appreciate your advice on which approach would be most suitable for accurately comparing the DNA damage profiles of these radionuclides. 

Thank you very much for your time and assistance. I look forward to your valuable guidance.

All the best,
Tai

### TOPAS-nBio category

Examples

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Attestation

- [X] I testify that I have fully read the bug reporting guideline beforehand.

**Comment by JoseRamosMendez** [comment]
Hi Tai, thanks for your interest in TOPAS-nBio. 
This implementation of DBSCAN works with ionization events in a homogenous water medium. So you don't need a DNA model. However, this version doesn't not consider indirect contributions from radiolytic species, which might be relevant as 177Lu decays in beta particles and gammas. So, you won't have the full picture.
Linear DNA is just a short segment of DNA, useful for simple test to help you understand what happens e.g., in a isolated spur of radiation, or different configurations of DNA nucleotide geometries, etc.
So, the way to go is the nucleus DNA model and step-by-step. Be aware that it'll be a bit expensive computationally,
Good luck.

**Reply by thanhtaiphys** [reply]
Dear Jose,

Thank you so much for answer my question.

I am currently trying to use the "DNADamageStepByStep" scorer instead of using DBSCAN in my simulation. Here is the setup of my model:

# Word
d:Ge/World/HLX			= 100 um
d:Ge/World/HLY			= 100 um
d:Ge/World/HLZ			= 100 um
s:Ge/World/Material			= "G4_WATER"

# WaterPhantom
s:Ge/WaterPhantom/Type		= "TsBox"
s:Ge/WaterPhantom/Parent	= "World"
s:Ge/WaterPhantom/Material	= "G4_WATER"
dc:Ge/WaterPhantom/HLX		= 80 um
dc:Ge/WaterPhantom/HLY		= 80 um
dc:Ge/WaterPhantom/HLZ		= 80 um

**#Cell**
s:Ge/Cell/Type= "TsSphere"
d:Ge/Cell/Rmin = 0. um
d:Ge/Cell/Rmax = 10. um
s:Ge/Cell/Parent = "WaterPhantom"
s:Ge/Cell/Material = "G4_WATER"

**#Nucleus**
s:Ge/Nucleus/Type= "TsSphere"
d:Ge/Nucleus/Rmin = 0. um
d:Ge/Nucleus/Rmax = 4 um
s:Ge/Nucleus/Parent = "Cell"
s:Ge/Nucleus/Material = "G4_WATER"
s:Ge/Nucleus/AssignToRegionNamed = "NucleusRegion"

**#DNA**
s:Ge/DNA/Type="TsCharltonDNA"
s:Ge/DNA/Parent="Nucleus"
s:Ge/DNA/Material="G4_WATER"
i:Ge/DNA/NumberOfBasePairs = 10

**#Source**
s:So/RadionuclideSource/Type                       = "Distributed"
s:So/RadionuclideSource/BeamParticle               = "GenericIon(71, 177)"
s:So/RadionuclideSource/Component                  = "DNA"

My purpose is to score **single-strand breaks (SSB) and double-strand breaks (DSB).** I am uncertain about which **component I should assign the source to**. I tried putting the source in **Component = "DNA**" and obtained results. However, when I changed the component to Nucleus, I got no output.
Could you please advise on the correct component to assign the source to for scoring SSBs and DSBs effectively? Besides that, based on my purpose, do I have to model Cell, Nucleus, and then DNA?

Thank you for your assistance.
Tai

**Reply by JoseRamosMendez** [reply]
Hi there, I would explore $TOPAS-nBio/examples/geometry/nucleusModel/ and create the particle source accordingly.

**Reply by thanhtaiphys** [reply]
Dear Jose,

Thank you for your guidance.

I am running an example as you suggested. 

**(1)** I want to score the output file as both "ASCII" and "root". Currently, I am using the following setting:
**s:Sc/DNADamageScorer/OutputType = "root"**

I tried changing it to:
**sv:Sc/DNADamageScorer/OutputType = 2 "root" "ascii"**

but the output was only a root file.

Could you please let me know how to score the output as both "ASCII" and "root"?

**(2)** Additionally, the default output file name is "topas.root". Could you please let me know how to customize the output file name to my desired name?

I tried setting it to:
**s:Sc/DNADamageScorer/OutputFile = "DNADamageHist10"
s:Sc/RootFileName = Sc/DNADamageScorer/OutputFile**
but the output was topas.root


Thank you for your help.
Tai

**Comment by JoseRamosMendez** [comment]
Hi there, the scorer is described in detail here https://nbio.readthedocs.io/en/latest/Scorers/DNAScorers.html. The parameter 

    s:Sc/RootFileName = "myfile"

should work, but maybe the way you define `OutputType` as a vector (must not be a vector) is conflicting.

**Comment by thanhtaiphys** [comment]
I can only score either a ROOT file or an ASCII file, but I cannot score both files simultaneously. Could you help me with this issue?

**Comment by thanhtaiphys** [comment]
When I replace the default source with the Lu-177 source, I encounter an error, and the simulation fails to run. Attached file is the input file. 
[ExampleNucleusDNADamage2.txt](https://github.com/user-attachments/files/16416209/ExampleNucleusDNADamage2.txt)

**Comment by JoseRamosMendez** [comment]
You **must** add physical processes accordingly to your simulation. Your plan is to use Lu-177 as a decay radionuclide, then you must include the "g4radioactivedecay" module.
Good luck!

**Comment by thanhtaiphys** [comment]
I have added some physics model but it doesn't work. For example:  sv:Ph/Default/Modules                      		= 4 "g4em-dna" "g4em-livermore" "g4decay" "g4radioactivedecay".
Please see the error below:
![image](https://github.com/user-attachments/assets/8bf5c4c8-ead4-40d1-aaf8-5b3ad5bf6967)

There is one thing I cannot explain. I ran it smoothly several times, but after my computer was restarted due to a power source issue, I ran the same input again, and it doesn't work. Do you mind running it on your PC to test?

**Reply by nknanfeng** [reply]
Hi Tai,

I hope this message finds you well.

I recently encountered a similar error while attempting to use a distributed particle source during the simulation, and I noticed you mentioned facing the same issue. Could you kindly share how you managed to resolve this problem? Does this error imply that the scoring module for the cell nucleus example does not support the "Distributed" type of particle source?

I would greatly appreciate your insights and assistance. Looking forward to your response.

Best regards,
Nanfeng

**Reply by thanhtaiphys** [reply]
Hi @nknanfeng 

I don't have an answer for you right now. I am still working on it, and there are many issues. I have tried with both sources, but as I mentioned above: "I ran it smoothly several times, but after my computer restarted due to a power source issue, I ran the same input again, and it didn't work." This means that sometimes it works, and sometimes it doesn't. It's difficult to explain!

**Comment by thanhtaiphys** [comment]
Dear @JoseRamosMendez 

Do you have time to run my input file? I have made many changes, but it still doesn't work.

Additionally, can TOPAS-nBio score Double-Strand Breaks (DSBs) as a function of LET? If so, could you provide some guidance on how to set this up.

Thank you so much!

**Reply by aoliaodaxiaoguaishou** [reply]
Dear @thanhtaiphys 

I hope this message finds you well.

I am currently working with the example files located in $TOPAS-nBio/examples/geometry/nucleusModel/ and have encountered some issues. Previously, I faced an error related to missing data files when running the examples. Fortunately, this issue has been resolved.

I would appreciate your assistance with running the modified examples or discussing any related topics concerning TOPAS-nBio. If you have any specific insights or recommendations, or if you are interested in collaborating on troubleshooting or further exploration, please let me know.

Thank you in advance for your help.

Best regards,
Lucy

**Comment by thanhtaiphys** [comment]
Hi Lucy @aoliaodaxiaoguaishou 

I am still working on it and don't have results yet. It's nice to work with you. Please send me email to thanhtaiphys@gmail.com if you want to work together.

All the best,
Tai
